contracts/secret-storage/SecretManager.sol

```diff
@@ -595,6 +596,72 @@ contract SecretManager is
         _replaceStore(_secretId, _enclaveAddress, enclaveIndex, true);
     }
 
+    function _renounceSecrets(
+        address _enclaveAddress
+    ) internal {
+        ( , address owner, , bool draining) = TEE_MANAGER.teeNodes(_enclaveAddress);
+        if(!draining)
+            revert SecretManagerStoreNotDraining();
+        // can use signature verification as well
+        if(_msgSender() != owner)
+            revert SecretManagerInvalidSecretOwner();
+
+        uint256 lastAliveTimestamp = SECRET_STORE.getSecretStoreLastAliveTimestamp(_enclaveAddress);
+        SECRET_STORE.renounceSecretsPreUpdate(_enclaveAddress, MARK_ALIVE_TIMEOUT, STAKING_PAYMENT_POOL);
+
+        // TODO: how to handle the newly assigned unacknowledged secrets
+        uint256[] memory storeAckSecretIds = SECRET_STORE.getStoreAckSecretIds(_enclaveAddress);
+        uint256 len = storeAckSecretIds.length;
+        uint256 occupiedStorage;
+        uint256 storageTimeUsage;
+        for (uint256 index = 0; index < len; index++) {
+            uint256 secId = storeAckSecretIds[index];
+            uint256 sizeLimit = userStorage[secId].sizeLimit;
+            occupiedStorage += userStorage[secId].sizeLimit;
+
+            storageTimeUsage += (_renounceSecret(_enclaveAddress, secId, lastAliveTimestamp) * sizeLimit);
+        }
+
+        uint256 usdcPayment = storageTimeUsage * SECRET_STORE_FEE_RATE;
+        USDC_TOKEN.safeTransfer(owner, usdcPayment);
+
+        SECRET_STORE.renounceSecretsPostUpdate(_enclaveAddress, occupiedStorage);
+    }
+
+    function _renounceSecret(
+        address _enclaveAddress,
+        uint256 _secretId,
+        uint256 _lastAliveTimestamp
+    ) internal returns (uint256 /* storageTimeDuration */) {
+        // uint256 secId = storeAckSecretIds[index];
+        uint256 enclaveIndex = _getSelectedEnclaveIndex(_secretId, _enclaveAddress);
+
+        uint256 ackTimestamp;
+        if(_isReplacedStore(_secretId, enclaveIndex))
+            ackTimestamp = userStorage[_secretId].selectedEnclaves[enclaveIndex].replacedAckTimestamp;
+        else
+            ackTimestamp = userStorage[_secretId].ackTimestamp;
+
+        uint256 startTimestamp = ackTimestamp > _lastAliveTimestamp ? ackTimestamp : _lastAliveTimestamp;
+        uint256 endTimestamp = userStorage[_secretId].endTimestamp;
+
+        _updateUsdcDepositPostPayment(_secretId, _enclaveAddress, enclaveIndex);
```

Comment:
there are redundant operations that this function is performing as well as L640 to L646. Can be improved?
like getting the ack time and last alive time.